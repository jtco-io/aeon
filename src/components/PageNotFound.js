// @flow
import React from 'react'
import { Link } from 'react-router-dom'

type Props = { location: { pathname: string } }
const PageNotFound = (props: Props) => (
  <div>
    <p>Sorry, no page found at {props.location.pathname}</p>
    <Link to="/">Go Home</Link>
  </div>
)

export default PageNotFound
