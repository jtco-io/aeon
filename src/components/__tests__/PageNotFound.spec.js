import React from 'react'
import PageNotFound from './../PageNotFound'
import { shallow } from 'enzyme'

test('it expects text with not found pathname', () => {
  const pathname = '/lost-page'
  const component = shallow(
    <PageNotFound location={{ pathname }}>Facebook</PageNotFound>,
  )

  expect(component).toMatchSnapshot()

  expect(component.find('p').text()).toContain(pathname)
})
