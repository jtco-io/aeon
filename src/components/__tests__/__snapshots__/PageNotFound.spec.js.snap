// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it expects text with not found pathname 1`] = `
<div>
  <p>
    Sorry, no page found at 
    /lost-page
  </p>
  <Link
    replace={false}
    to="/"
  >
    Go Home
  </Link>
</div>
`;
